test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -4533,7 +4533,7 @@ BOOST_AUTO_TEST_CASE(no_warn_about_callcode_as_local)
 	char const* text = R"(
 		contract test {
 			function callcode() {
-				var x = this.callcode;
+				test.callcode();
```

Comment:
What is the point of this test? The test case name says "as local".

Reply:
We warn when using `address(this).callcode()`, but we should not warn when there is an actual function called `callcode`.

Reply:
I think we should update the test case name to reflect that (it is confusing a bit now).